
![Static Badge](https://img.shields.io/badge/language-python-780000) ![Static Badge](https://img.shields.io/badge/language-JupyterNotebook-FDF0D5) ![Static Badge](https://img.shields.io/badge/license-UCSD-C1121F) ![Static Badge](https://img.shields.io/badge/tools-VarScan-669BBC) 

# SNP Scout

This is a demonstration project for CSE 185. `SNP Scout` implements a simpler version of the SNP caller available through VarScan. See the [VarScan](https://varscan.sourceforge.net/using-varscan.html) page for more details.  
# Install Instructions
No additional libraries are required for installation. 
To install `SNP Scout`, you can use the following command:
# Basic Usage
The basic usage of `SNP Scout` is given by the following:
EDIT
# SNP Scout Options
The only required input to `SNP Scout` is a SAMtools pileup file. Users can specify additional options below:
* `-m FREQ`, `--min_var_freq FREQ`: minimum variant allele frequency threshold. By default, 0.01 is used.
* `-c COV`, `--min_coverage COV`: minimum read depth at a position to make a call for variants. By default, 8 is used.
* `-o FILE`, `--out FILE`: write output to the specified file. Otherwise, output is written to the terminal.   
# File Format 
The output file format is the same as the VarScan variant calling method. See the [VarScan](https://varscan.sourceforge.net/using-varscan.html) page for more details.
# Contributors 
This repository was generated by Sally Ha, Mufei Li, and Margaret Jones...(EDIT)
